x-pack/legacy/plugins/apm/server/lib/transactions/breakdown/index.ts

```diff
@@ -135,11 +136,12 @@ export async function getTransactionBreakdown({
     return breakdowns;
   };
 
-  const visibleKpis = sortByOrder(
-    formatBucket(resp.aggregations),
-    'percentage',
-    'desc'
-  ).slice(0, MAX_KPIS);
+  const visibleKpis = resp.aggregations
+    ? sortByOrder(formatBucket(resp.aggregations), 'percentage', 'desc').slice(
+        0,
+        MAX_KPIS
+      )
+    : [];
```

Comment:
Can you avoid the ternary like this?
```js
const visibleKpis = sortByOrder(formatBucket(resp.aggregations || []), 'percentage', 'desc').slice(0, MAX_KPIS)
```

Reply:
`resp.aggregations` is not an array, but a plain object. If I pass in a plain object, TypeScript will complain about a type mismatch.

Reply:
Ahh, yeah. You could also change `formatBucket` like:
```ts
  const formatBucket = (
    aggs:
      | typeof resp['aggregations']
      | Required<typeof resp>['aggregations']['by_date']['buckets'][0]
  ) => {
    if (!aggs) {
      return [];
    }
```

But if you prefer the ternary that's also totally fine.